cluster.py

```diff
@@ -25,6 +50,13 @@ def __init__(self, driver_directory: Path, version: str, configuration: Dict[str
         self._cluster.populate(1)
         logger.info("Cluster prepared")
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.remove()
+        release_cluster_lock(self._cluster_lock)
```

Comment:
Can you please use `self._cluster_lock.close()` directly?

Reply:
see above